Adds new Design System SearchBarcomponent

martins87 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Link to Issue
Closes: #4119 

## Description of Changes
- Adds new Design System to SearchBar component
- Adds Community Tag when on some community. The tag means all search will be within this community.
- `'Backspace'` key removes tag (and community filter)
- `'Escape'` key resets input
- Added useSearchResults hook. Receives filters to search on: 'threads', 'replies', 'communities', 'members'. Can also receive an optional number of results per filter.

jnaviask commented:
@martins87 please fix linter errors

masvelio commented on packages/commonwealth/.storybook/stories/components/SearchBar.stories.tsx:
```diff
@@ -0,0 +1,47 @@
+const searchbar = {
+  title: "Components/SearchBar",
+  component: CWSearchBar,
+} satisfies Meta<typeof CWSearchBar>;
```
curious what is the reason we use `satisfies`? I remember that I removed it some time ago and everything seemed to work fine?

martins87 replied:
Removed. I just added because it was on Storybook docs as reference.

masvelio commented on packages/commonwealth/client/scripts/views/components/component_kit/cw_component_showcase.tsx:
```diff
@@ -11,6 +11,7 @@ import { CWBreadcrumbs } from './cw_breadcrumbs';
 import { CWButton } from './new_designs/cw_button';
 import { CWUpvote } from './new_designs/cw_upvote';
 import { CWThreadAction } from './new_designs/cw_thread_action';
+import { CWSearchBar } from './new_designs/cw_search_bar';
```
we try to ditch `snake_case` for file names. Please use `CWSearchBar`

martins87 replied:
Addressed.

masvelio commented on packages/commonwealth/client/scripts/views/components/component_kit/new_designs/cw_search_bar.tsx:
```diff
@@ -0,0 +1,130 @@
+
+import { ComponentType } from './../types';
+
+import 'components/component_kit/new_designs/cw_search_bar.scss';
```
as well here `CWSearchBar.scss`

martins87 replied:
Also with the .scss files. Thanks for the tip on using it on the next files/components.

masvelio commented on packages/commonwealth/client/scripts/views/components/component_kit/new_designs/cw_search_bar.tsx:
```diff
@@ -0,0 +1,130 @@
+
+const Tag: FC<TagProps> = ({ communityName, disabled, onClick }) => {
+  const handleClick = () => {
+    console.log('clicked!');
```
remove console log

martins87 replied:
Addressed.

masvelio commented on packages/commonwealth/client/scripts/views/components/component_kit/new_designs/cw_search_bar.tsx:
```diff
@@ -0,0 +1,130 @@
+  onClick: () => void;
+};
+
+const Tag: FC<TagProps> = ({ communityName, disabled, onClick }) => {
```
this tag should be in separate file. Could be `CWTag` component

martins87 replied:
Addressed as suggested.

masvelio commented on packages/commonwealth/client/scripts/views/components/component_kit/new_designs/cw_search_bar.tsx:
```diff
@@ -0,0 +1,130 @@
+      <div className="name">
+        <CWCommunityAvatar
+          size="small"
+          // community={app.chain.meta}
```
remove comment

martins87 replied:
Addressed.

masvelio commented on packages/commonwealth/client/scripts/views/components/component_kit/new_designs/cw_search_bar.tsx:
```diff
@@ -0,0 +1,130 @@
+          size="small"
+          // community={app.chain.meta}
+          community={null}
+          onClick={() => {}}
```
I dont think we should handle click on avatar. Only click on tag matters here

martins87 replied:
Addressed. Indeed it only makes sense to handle clicks on the "X" icon.

masvelio commented on packages/commonwealth/client/scripts/views/components/component_kit/new_designs/cw_search_bar.tsx:
```diff
@@ -0,0 +1,130 @@
+  };
+
+  return (
+    <div className="tag">
```
capitalise parent class `Tag`

martins87 replied:
Addressed.

masvelio commented on packages/commonwealth/client/scripts/views/components/component_kit/new_designs/cw_search_bar.tsx:
```diff
@@ -0,0 +1,130 @@
+      )}
+    >
+      <MagnifyingGlass className="magnifyingGlass" weight="regular" size={24} />
+      {tag && <Tag communityName="1inch" onClick={() => setTag(false)} />}
```
community name should be a prop, not hardcoded

martins87 replied:
Addressed. I didn't add multiple tags as suggested by Meesh. Still to be implemented the search on a specific community (shown as a tag).

masvelio reviewed (changes requested):
[as pointed by](https://github.com/hicommonwealth/commonwealth/issues/4119#issuecomment-1603016921) @zakhap we should use mui-autocomplete for this. 
Generally, this should be fully functional component meaning that it should support eg:
- showing search list during typing,
- Maybe selecting multiple tags
- adding tag on hitting enter

We should not build components from scratch especially those that are complicated, because there are way too many edge cases that have to be handled. We should always use mui-base and adjust styles/logic for our needs

masvelio commented on packages/commonwealth/client/scripts/views/components/component_kit/new_designs/cw_search_bar.tsx:
```diff
@@ -0,0 +1,130 @@
+    setSearchTerm(value);
+  };
+
+  const handleOnInput = (e) => {
+    if (e.currentTarget.value?.length === 0) {
+      setIsTyping(false);
+    } else {
+      setIsTyping(true);
+    }
```
the styling should be done with onFocus and onBlur, not based on typing

martins87 replied:
Not yet handled.

martins87 replied:
The styling here is different for the states "focus" and "typing". On focus, the border turns blue, when typing there is also the blue shadow.

Currently the shadow is displayed when the state goes to "isTyping". What are your thoughts on this?

jnaviask commented:
@martins87 please reply to Marcin's comments noting whether the raised issue has been addressed, and if not then why

martins87 commented on packages/commonwealth/client/styles/components/component_kit/new_designs/cw_search_bar.scss:
```diff
@@ -0,0 +1,92 @@
+
+}
+
+.tag {
```
Addressed.

martins87 commented on packages/commonwealth/client/styles/components/component_kit/new_designs/cw_search_bar.scss:
```diff
@@ -0,0 +1,92 @@
+
+.inputElement {
+  flex: 1;
+}
```
Addressed.

martins87 commented on packages/commonwealth/client/scripts/views/components/component_kit/new_designs/cw_search_bar.tsx:
```diff
@@ -0,0 +1,130 @@
+export const CWSearchBar = (props: SearchBarProps) => {
+  const [searchTerm, setSearchTerm] = useState<string>('');
+  const [isTyping, setIsTyping] = useState<boolean>(false);
+  const [tag, setTag] = useState<boolean>(true);
```
Removed.

martins87 commented:
> @martins87 please reply to Marcin's comments noting whether the raised issue has been addressed, and if not then why

Done.

masvelio commented on packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWTag.tsx:
```diff
@@ -0,0 +1,44 @@
+  return (
+    <div className="SearchBarTag">
+      <div className="name">
+        <CWCommunityAvatar size="small" community={null} />
```
how I am supposed to display the community avatar from outside of the component? this should not be `null` -- it should be a prop passed from the parent, similarly to `communityName`

martins87 replied:
Addressed.

masvelio commented on packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWSearchBar.tsx:
```diff
@@ -0,0 +1,148 @@
+        />
+        {communityName && (
+          <CWTag
+            communityName={communityName}
```
avatar is missing here

martins87 replied:
Addressed.

masvelio commented on packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWSearchBar.tsx:
```diff
@@ -0,0 +1,148 @@
+  InputInternalStyleProps &
+  React.HTMLAttributes<HTMLDivElement>;
+
+const communities = [
+  'Altitude',
+  'Terra Classic',
+  'Osmosis',
+  'Qwoyn Network',
+  '1inch',
+  'Stargate Finance',
+  'Timeless',
+  'Terra Agora',
+  'Injective',
+  'Juno',
+  'Common',
+];
```
this variable cannot be defined here. see the comment below

martins87 replied:
Removed.

masvelio commented on packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWSearchBar.tsx:
```diff
@@ -0,0 +1,148 @@
+  'Common',
+];
+
+export const CWSearchBar: FC<SearchBarProps> = ({ disabled, placeholder }) => {
```
again, how would you like to manage the data of this component from outside of the component? You are missing props that will handle that eg:
- you don't want to have the list of communities defined in this file. You want to fetch it from the API and pass it to the searchbar component by props eg `options`
- if you select some community you would like to react on that outside of the component, probably filter the threads by this community, so you need to expose `selectedItem` outside of this component
- also, this CWSearchBar and CWTag should not be bounded to the community, they should be generic, so we should use `options`, `items`, `selectedItem` itd instead of `communityName`, `community` etc

martins87 replied:
The list of communities (chains) is now passed as props to the component as follows
```typescript
<CWSearchBar
  options={app.config.chains.getAll()}
  placeholder="Search Common"
/>
```

`seletecItem` yet to be implemented.

masvelio reviewed (changes requested):
It still needs some work on the code, let me see how the component behaves

masvelio commented:
1. when I click inside the input but very close to the border, the focus is not being activated

https://github.com/hicommonwealth/commonwealth/assets/14819225/6ad91d2c-4f54-44f7-bab2-b74f974df3a9


2. active border (light blue) is activated only when I start typing. Shouldn't it be activated on input focus? Also when I type something (or select the item) and then click outside, the active state does not go away

https://github.com/hicommonwealth/commonwealth/assets/14819225/ba3aafd6-4c5f-463d-bfa5-907769914383

3. probably it would be nice if in this state I hit backspace, the selected item is removed

![image](https://github.com/hicommonwealth/commonwealth/assets/14819225/e976b15e-7b76-4ac9-afe0-72f0766db718)

4. It is quite common to use keyboard arrows to operate this type of search (up & down) and enter to select the item. Currently it is not supported.

5. We don't have any indicator if the typed value has not march -- something like `no options` in the dropdown

All of these above are just my thoughts that are usually implemented in this kind of component. You can play around with the searchbox from mui to have a sense of the good UX [here](https://mui.com/material-ui/react-autocomplete/#multiple-values). I am not saying though we need to implement all of these in this PR. This is a decision for @zakhap & @meeshlin

zakhap commented:
Of the issues @masvelio brought up above, we should resolve 1, 2, + 5.

> 5.  We don't have any indicator if the typed value has not march -- something like no options in the dropdown

"No results found" should be the copy.

Q: Does the new CWTag interfere with the tag would @Miaplacidus was doing?

jnaviask commented:
@martins87 please fix typecheck issue

martins87 commented on packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWSearchBar.tsx:
```diff
@@ -0,0 +1,148 @@
+import React, { FC, ChangeEvent, useState, useEffect } from 'react';
```
Addressed.

martins87 commented on packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWTag.tsx:
```diff
@@ -0,0 +1,44 @@
+  };
+
+  return (
+    <div className="SearchBarTag">
```
Addressed.

martins87 commented:
> 

1. I'm not getting this behaviour when clicking close to the border. Currently when hovered inside the component, the border turns blue.
Testing it on localhost, because storybook was broken when ChainInfo was added. Will check this again later on.

2. As per Figma designs, blue shadow border should be activated only when typing. State is now off when clicked away.

3. Addressed.

4. Yet to be implemented.

5. Addressed.

martins87 commented:
> 

1. I'm not getting this behaviour when clicking close to the border. Currently when hovered inside the component, the border turns blue.
Testing it on localhost, because storybook was broken when ChainInfo was added. Will check this again later on.

2. As per Figma designs, blue shadow border should be activated only when typing. State is now off when clicked away.

3. Addressed.

4. Yet to be implemented.

5. Addressed.

martins87 commented:
> Of the issues @masvelio brought up above, we should resolve 1, 2, + 5.
> 
> > 5. We don't have any indicator if the typed value has not march -- something like no options in the dropdown
> 
> "No results found" should be the copy.
> 
> Q: Does the new CWTag interfere with the tag would @Miaplacidus was doing?

Please check number list comments below and let me know your thoughts.

I don't think the CWTag interfere on the one you mentioned because it receives the chain as props and displays chain name and icon.

masvelio commented:
@martins87 

> 2. As per Figma designs, blue shadow border should be activated only when typing. State is now off when clicked away.
No, there is no state for typing as I pointed out in the other comment. It should have border on hover and border+shadow on focus. Thats it

> 4. Yet to be implemented.
Zak confirmed that you do not need to implement it

masvelio commented on packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWTag.tsx:
```diff
@@ -0,0 +1,45 @@
+  };
+
+  return (
+    <div className="CWTag">
```
Ifu implemented tag already recently. You should use it instead of implementing your own

martins87 replied:
I made some changes on Ifu's tag for the CWSearchBar to use it. Please let me know your thoughts.

masvelio commented on packages/commonwealth/client/styles/components/component_kit/new_designs/CWSearchBar.scss:
```diff
@@ -0,0 +1,155 @@
+
+.option {
+  list-style: none;
+  padding: 8px;
+  border-radius: 8px;
+  cursor: default;
+
+  display: flex;
+  flex-direction: row;
+  gap: 8px;
+
+  font-family: 'NeueHaasUnica';
+  font-style: normal;
+  font-weight: 400;
+  font-size: 14px;
+  line-height: 20px;
+  align-items: center;
+  letter-spacing: 0.01em;
+  font-feature-settings: 'tnum' on, 'lnum' on;
+
+  &:hover {
+    cursor: pointer;
+    background-color: $neutral-100;
+  }
+}
+
+.noResults {
+  position: absolute;
+  top: 100%;
+  left: 0;
+  width: 100%;
+  margin-top: 0px;
+  padding: 14px;
+  max-height: 300px;
+  border-radius: 6px;
+  background-color: $white;
+  border: 1px solid #d0d7de;
+  color: #24292f;
+  box-shadow: 0px 4px 30px #d0d7de;
+}
```
I think this is 6th time I put the same comment. All of these classes should be encapsulated because you are affecting other `container` and `option` classes

martins87 replied:
Addressed.

masvelio commented on packages/commonwealth/client/styles/components/component_kit/new_designs/CWSearchBar.scss:
```diff
@@ -0,0 +1,155 @@
+  background-color: $white;
+  border: 1px solid #d0d7de;
+  color: #24292f;
+  box-shadow: 0px 4px 30px #d0d7de;
```
we should not use hardcoded hex, only defined colors

martins87 replied:
Addressed.

CowMuon replied:
Question: Why did we hard code the hex here? That doesn't seem to align with our development, so just tryna understand how that got in there.

martins87 commented:
> @martins87
> 
> > 2. As per Figma designs, blue shadow border should be activated only when typing. State is now off when clicked away.
> 
> No, there is no state for typing as I pointed out in the other comment. It should have border on hover and border+shadow on focus. Thats it
> 
> > 4. Yet to be implemented.
> 
> Zak confirmed that you do not need to implement it

2. Indeed. It has been addressed.

4. Ok.

martins87 commented:
Note: the component is working fine on /components.
Its story on storybook although is broken because of some error with ChainInfo importing. I'll dig into this.

zakhap commented on packages/commonwealth/client/scripts/views/components/component_kit/new_designs/cw_tag.tsx:
```diff
@@ -30,15 +33,25 @@ export type TagProps = {
+  trimAt,
+  classNames,
+  loginIcon,
+  community,
```
This prop should be optional, no?

martins87 replied:
It is. It's defined on TagProps type.

zakhap commented:
Behavior for search bar:
- If in community, automatically show the community tag for filtering search results.
  - Tag can be x’d out, which removes the filter from the search query/results.
    - The tag cannot be recreated by the user.
    - the search field is only for the search query, not for deciding/inputing filters (like by community).
- If not in a community, the community tag should not be present
- If the tag is present, do not search for or display communities. All other content should be shown: comments, threads, etc.
- If the tag is not present, include communities + all other items in the dropdown

CowMuon commented:
Late to the party here, but the very first thing I'm noticing is that the PR description here isn't even filled out correctly. @martins87 what happened?

martins87 commented:
> Late to the party here, but the very first thing I'm noticing is that the PR description here isn't even filled out correctly. @martins87 what happened?

Done.

martins87 commented:
> 

All addressed.

martins87 commented:
> Behavior for search bar:
> 
> * If in community, automatically show the community tag for filtering search results.
>   
>   * Tag can be x’d out, which removes the filter from the search query/results.
>     
>     * The tag cannot be recreated by the user.
>     * the search field is only for the search query, not for deciding/inputing filters (like by community).
> * If not in a community, the community tag should not be present
> * If the tag is present, do not search for or display communities. All other content should be shown: comments, threads, etc.
> * If the tag is not present, include communities + all other items in the dropdown

All addressed.

rbennettcw commented on packages/commonwealth/client/scripts/hooks/useSearchResults.ts:
```diff
@@ -0,0 +1,66 @@
+    orderBy: APIOrderBy.Rank,
+    orderDirection: APIOrderDirection.Desc,
+  };
+  const queryEnabled = debouncedSearchTerm.length > 0;
+
+  const { data: threadsData } = useSearchThreadsQuery({
+    ...sharedQueryOptions,
+    enabled: queryEnabled,
+  });
+
+  const { data: commentsData } = useSearchCommentsQuery({
+    ...sharedQueryOptions,
+    enabled: queryEnabled,
+  });
+
+  const { data: chainsData } = useSearchChainsQuery({
+    ...sharedQueryOptions,
+    enabled: queryEnabled,
+  });
+
+  const { data: profilesData } = useSearchProfilesQuery({
+    ...sharedQueryOptions,
+    includeRoles: false,
+    enabled: queryEnabled,
+  });
+
+  const searchResults = useMemo(() => {
+    return {
+      [SearchScope.Threads]: filters.includes("threads") ? threadsData?.pages?.[0]?.results || [] : [],
+      [SearchScope.Replies]: filters.includes("replies") ? commentsData?.pages?.[0]?.results || [] : [],
+      [SearchScope.Communities]: filters.includes("communities") ? chainsData?.pages?.[0]?.results || [] : [],
+      [SearchScope.Members]: filters.includes("members") ? profilesData?.pages?.[0]?.results || [] : [],
+    };
+  }, [threadsData, chainsData, profilesData, commentsData, filters]);
```
Here, it's fetching __all__ results (threads, comments, chains, profiles), but then returning only some results, based on filter.

For efficiency, the `enabled` option should be set based on the filter, so that it's not fetching data that won't be returned/rendered.

martins87 replied:
[Addressed](https://github.com/hicommonwealth/commonwealth/commit/046a7bc5daa2fcc8a7bdf8c520ef80fe3ab8f6fa).

masvelio commented on packages/commonwealth/client/scripts/views/components/component_kit/cw_component_showcase.tsx:
```diff
@@ -1213,6 +1224,10 @@ export const ComponentShowcase = () => {
       </div>
+      <div className="searchbar-gallery">
+        <CWText type="h3">SearchBar</CWText>
+        <CWSearchBar placeholder="Search Common" />
```
`Search Common` is a default value for placeholder prop, so you dont need to pass it here

martins87 replied:
Addressed.

masvelio commented on packages/commonwealth/client/scripts/hooks/useSearchResults.ts:
```diff
@@ -0,0 +1,66 @@
+
+  const searchResults = useMemo(() => {
+    return {
+      [SearchScope.Threads]: filters.includes("threads") ? threadsData?.pages?.[0]?.results || [] : [],
+      [SearchScope.Replies]: filters.includes("replies") ? commentsData?.pages?.[0]?.results || [] : [],
+      [SearchScope.Communities]: filters.includes("communities") ? chainsData?.pages?.[0]?.results || [] : [],
+      [SearchScope.Members]: filters.includes("members") ? profilesData?.pages?.[0]?.results || [] : [],
```
if you do `pages[0]`, does it mean we always display only first page?

martins87 replied:
I did a test passing a large number to [useSearchResults hook](https://github.com/hicommonwealth/commonwealth/blob/daniel.4119.search-bar/packages/commonwealth/client/scripts/hooks/useSearchResults.ts), searching for the term `'a'` and these are the results brought:

![image](https://github.com/hicommonwealth/commonwealth/assets/30223098/f58d8666-a54c-4b0c-b6d4-f00459738654)

![image](https://github.com/hicommonwealth/commonwealth/assets/30223098/33a65ba4-f6b6-4f74-9345-f14f6d0dc8bb)

So the results for every section (i.e. 'threads', 'replies', 'communities' or 'members') are inside `pages[0]`.
It seems that this data structure was designed like this. Here is another example of the results being read from `data.pages[0].results` array: https://github.com/hicommonwealth/commonwealth/blob/master/packages/commonwealth/client/scripts/views/pages/manage_community/ManageCommunityPage.tsx#L64

masvelio commented on packages/commonwealth/client/scripts/views/components/component_kit/new_designs/search/dropdown.tsx:
```diff
@@ -0,0 +1,241 @@
+  searchResults: any;
+};
+
+const SearchBarThreadPreviewRow = (props: SearchBarThreadPreviewRowProps) => {
```
this is not aligned with our new standards described here
https://github.com/hicommonwealth/commonwealth/blob/master/wiki/React-Best-Practices-And-Improvements.md#directory-structure

- each component in own file
- each file has its own css file with the same name, and the top className aligned with file/component name
- component name is the same as file name
- all of components should be imported to and exported from index file

martins87 replied:
Addressed.

masvelio replied:
looks much better!

masvelio commented on packages/commonwealth/client/styles/components/component_kit/new_designs/CWSearchBar.scss:
```diff
@@ -0,0 +1,215 @@
+  }
+}
+
+.SearchBarCommentPreviewRow {
```
look at my other comment about having separate css files for each component

martins87 replied:
Addressed.

masvelio reviewed (commented):
please see the comments for code changes.
Also would be good to push it to the demo right away and let Jess know to test it out

martins87 commented:
> please see the comments for code changes. Also would be good to push it to the demo right away and let Jess know to test it out

Done. I've tagged Jessica to review it.

masvelio commented on packages/commonwealth/client/scripts/views/components/react_quill_editor/react_quill_editor.tsx:
```diff
@@ -236,51 +242,67 @@ const ReactQuillEditor = ({
+                        value={contentDelta}
+                        onFocus={() => setIsFocused(true)}
+                        onBlur={() => setIsFocused(false)}
+                        onChange={handleChange}
+                        onChangeSelection={(selection: RangeStatic) => {
+                          if (!selection) {
+                            return;
+                          }
+                          lastSelectionRef.current = selection;
+                        }}
+                        formats={isMarkdownEnabled ? [] : undefined}
+                        modules={{
+                          toolbar: {
+                            container: `#${toolbarId}`,
+                            handlers: isMarkdownEnabled
+                              ? markdownToolbarHandlers
+                              : undefined,
+                          },
+                          imageDropAndPaste: {
+                            handler: handleImageDropAndPaste,
+                          },
+                          clipboard: {
+                            matchers: clipboardMatchers,
+                          },
+                          mention,
+                          magicUrl: !isMarkdownEnabled,
+                          keyboard: isMarkdownEnabled
+                            ? markdownKeyboardShortcuts
+                            : undefined,
+                          imageUploader: {
+                            upload: handleImageUploader,
+                          },
+                        }}
+                      />
+                    </div>
+                    {provided.placeholder}
+                  </div>
+                )}
+              </Droppable>
+            </DragDropContext>
```
these commits should not be part of this PR

martins87 replied:
Removed.

masvelio reviewed (approved):
codewise looks good, approving 👍 

Things TODO before merge:
- [x] resolve conflicts
- [ ] get design signoff
- [ ] get rid of drag&drop commits/code from this PR

martins87 commented:
> codewise looks good, approving 👍
> 
> Things TODO before merge:
> 
> * [x]  resolve conflicts
> * [ ]  get design signoff
> * [x]  get rid of drag&drop commits/code from this PR

Cc @jessmart1213

martins87 commented on packages/commonwealth/client/scripts/views/components/component_kit/new_designs/SearchBar/CWSearchBar.scss:
```diff
@@ -0,0 +1,103 @@
+@import '../../../../../../styles/shared';
+
+@mixin everyRowStyles {
+  cursor: pointer;
+  padding: 8px;
+
+  &:hover {
+    background-color: $neutral-100;
+    border-radius: 6px;
+  }
+}
```
[Addressed](https://github.com/hicommonwealth/commonwealth/pull/4462/commits/dba9667af1c8c29f0c9c13aba1505118df2ee2fd).

jessmart1213 commented:
### Design QA feedback

Hi @martins87 I just reviewed updates and everything looks great! Thank you!

I do have an additional piece of feedback that I did not catch in the first Design QA round, which was that the updated search bar has a fixed width at all breakpoints, when it should be responsive. This is important because a search bar with a fixed width blocks user functionality at smaller breakpoints, as the search pushes the utility nav (notification cta, profile dropdown, etc) out of view. 

Expected responsive behavior is what we have currently implemented. Please see attached screen recording for more context. 

https://github.com/hicommonwealth/commonwealth/assets/133259695/8e794e8c-a18a-4160-aedc-2b9966ce976c

I'm not sure if the search bar being responsive was part of the acceptance criteria for this ticket, so lmk if this requires a separate ticket!

martins87 commented:
> ### Design QA feedback
> Hi @martins87 I just reviewed updates and everything looks great! Thank you!
> 
> I do have an additional piece of feedback that I did not catch in the first Design QA round, which was that the updated search bar has a fixed width at all breakpoints, when it should be responsive. This is important because a search bar with a fixed width blocks user functionality at smaller breakpoints, as the search pushes the utility nav (notification cta, profile dropdown, etc) out of view.
> 
> Expected responsive behavior is what we have currently implemented. Please see attached screen recording for more context.
> 
>  Screen.Recording.2023-08-29.at.6.50.51.PM.mov 
> I'm not sure if the search bar being responsive was part of the acceptance criteria for this ticket, so lmk if this requires a separate ticket!

Addressed. Deployed to [demo](https://commonwealth-demo.herokuapp.com/theresearchcow/discussions) for testing.

Cc @jessmart1213 @jnaviask

masvelio commented:
@jnaviask @martins87 @jessmart1213 lets make sure that this will be merged before we start working on Gating. This component will be needed in [Groups Display Page](https://www.figma.com/file/ePuGFatYQI6GL72i27sVwj/Gating?type=design&node-id=794%3A28626&mode=design&t=gyL94wDcdNGo0nre-1)

jessmart1213 commented:
The search bar component has Design sign-off! All my requested updates have been met. Looks great @martins87! cc @masvelio

martins87 commented:
> The search bar component has Design sign-off! All my requested updates have been met. Looks great @martins87! cc @masvelio

@jnaviask
